### PR TITLE
BookManagement_CRUD: Basic book management using in-memory database with create, read, update, delete functions

### PR DESCRIPTION
**Pull request by vuong196**

### CODE REVIEW

File: src/AddFormServlet.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import java.io.IOException;
+import java.io.PrintWriter;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
```
Comment (hongvo2308) [inline src/AddFormServlet.java]: Import order follow the alphabet

File: src/AddFormServlet.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+@WebServlet("/AddFormServlet")
+public class AddFormServlet extends HttpServlet {
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = 1L;
```
Comment (ces-hongvo) [inline src/AddFormServlet.java:14]: Private static variable should be named like this _SERIAL_VERSION_UID

File: src/AddFormServlet.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+	protected void processRequest(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        response.setContentType("text/html;charset=UTF-8");
+
```
Comment (ces-hongvo) [inline src/AddFormServlet.java:19]: please adjust the indent

File: src/AddServlet.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
+
+import java.io.IOException;
+import java.io.PrintWriter;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
```
Comment (hongvo2308) [inline src/AddServlet.java]: Wrong import order

File: src/AddFormServlet.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+@WebServlet("/AddFormServlet")
+public class AddFormServlet extends HttpServlet {
+	/**
+	 *
```
Comment (ces-hongvo) [inline src/AddFormServlet.java:12]: Remove unused comments

File: src/AddServlet.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+@WebServlet("/AddServlet")
+public class AddServlet extends HttpServlet {
+	/**
+	 *
```
Comment (ces-hongvo) [inline src/AddServlet.java:14]: Remove this

File: src/AddServlet.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/AddServlet")
+public class AddServlet extends HttpServlet {
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = 1L;
+
```
Comment (ces-hongvo) [inline src/AddServlet.java:17]: Follow the convention I have mentioned above

File: src/AddServlet.java [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+	protected void processRequest(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        response.setContentType("text/html;charset=UTF-8");
+
```
Comment (ces-hongvo) [inline src/AddServlet.java:21]: Adjust the Indent

File: src/AddServlet.java [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+		processRequest(request, response);
+		
+		PrintWriter out=response.getWriter();
+		
+		String name=request.getParameter("name");
+		String author=request.getParameter("author");
+		Book book = new Book(name, author);
+		
+		int status = BookDAO.save(book);
+
```
Comment (ces-hongvo) [inline src/AddServlet.java:35]: Please use try - catch when using the Service function.
Catching with the log to help the debug process

File: src/Book.java [hunk 10]
```diff
@@ -0,0 +1,42 @@
+import java.util.UUID;
+
+public class Book {
+	private String id;
+	private String name;
+	private String author;
+
```
Comment (ces-hongvo) [inline src/Book.java:7]: the private attribute should be named: _id, _name, _author

File: src/AddFormServlet.java [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+	 */
+	private static final long serialVersionUID = 1L;
+
+	protected void processRequest(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+        response.setContentType("text/html;charset=UTF-8");
+        
+    }
+	
+	protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
```
Comment (ces-hongvo) [inline src/AddFormServlet.java:22]: Adding the annotation @Override if it is override

File: src/BookDAO.java [hunk 12]
```diff
@@ -0,0 +1,61 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class BookDAO {
+	private static List<Book> list = new ArrayList<Book>(){/**
+		 * 
+		 */
```
Comment (ces-hongvo) [inline src/BookDAO.java:7]: remove unused comments

File: src/BookDAO.java [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+		private static final long serialVersionUID = 1L;
+
+	{
+		 add(new Book("The Maze Runner", "James Dashner"));
```
Comment (ces-hongvo) [inline src/BookDAO.java:11]: adjust the indent here

File: src/BookDAO.java [hunk 14]
```diff
@@ -0,0 +1,61 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class BookDAO {
+	private static List<Book> list = new ArrayList<Book>(){/**
+		 *
```
Comment (ces-hongvo) [inline src/BookDAO.java:6]: Naming "list" is unclear, change to "books" or "bookRepository" instead

File: src/BookDAO.java [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+		list.add(book);
+		status = 1;
+		return status;
+	}
+    
+    public static int update(Book book){
+		int status=0;
+
+		for(Book b:list)
+			if (book.getId().equals(b.getId())){
```
Comment (ces-hongvo) [inline src/BookDAO.java:27]: try to use { after for

File: src/BookDAO.java [hunk 16]
```diff
@@ -0,0 +1,61 @@
...
+				status = 1;
+				return status;
+			}
+		}
+		return status;
+	}
+    
+    public static Book getBookById(String id){
+    	for(Book b:list)
+			if (b.getId().equals(id))
```
Comment (ces-hongvo) [inline src/BookDAO.java:52]: try to use {} after for

File: src/BookDAO.java [hunk 17]
```diff
@@ -0,0 +1,61 @@
...
+				System.out.println(id);
+				list.remove(i);
+				status = 1;
+				return status;
+			}
+		}
+		return status;
+	}
+    
+    public static Book getBookById(String id){
```
Comment (ces-hongvo) [inline src/BookDAO.java:50]: add a space between ) {

**Review by ces-hongvo — CHANGES_REQUESTED** [review]
Hi,
I have made some comments.
Please update them then I will look at them again.
Thank you.

**Pull request merged**